### PR TITLE
Plans: Adapt Jetpack Free card so its code can be re-used

### PR DESCRIPTION
**Pull request by mattgawarecki**
Relates to `1200412004370260-as-1200532973279652`.

#### Changes proposed in this Pull Request

* Extract the core implementation of `JetpackFreeCard` and `JetpackFreeCardButton` into a new component, `ProductWithoutPrice`, that can be re-used for Jetpack CRM Free (or other free products in the future).
* Re-write `JetpackFreeCard` to use `ProductWithoutPrice`.
* Move `JetpackFreeCard` under the `my-sites/` hierarchy, since that's the only place it's used.
* Re-write the existing unit tests from `JetpackFreeCardButton` so that they test the `useJetpackFreeButtonProps` hook.

#### Testing instructions

* Evaluate styles, implementation, and unit test code, ensuring they're readable and intuitive.
* Verify that in all Calypso experiences, the "Get started for free" button on the Jetpack Free product card looks and behaves exactly the same as in production.
* Verify that all unit tests still pass.

### CODE REVIEW

File: client/components/jetpack/card/product-without-price/style.scss [hunk 1]
```diff
@@ -1,9 +1,12 @@
 @import '@wordpress/base-styles/breakpoints';
 @import '@wordpress/base-styles/mixins';
 
-.jetpack-free-card {
+.product-without-price,
+.product-without-price--full-width {
```
Comment (mattgawarecki) [inline client/components/jetpack/card/product-without-price/style.scss:5]: This `--full-width` variant will become important as I build the PR to add Jetpack CRM Free to the bottom grid row.

File: client/my-sites/plans/jetpack-plans/types.ts [hunk 2]
```diff
@@ -57,6 +57,7 @@ export type PlanGridProducts = {
 };
 
 export interface JetpackFreeProps {
+	fullWidth?: boolean;
```
Comment (mattgawarecki) [inline client/my-sites/plans/jetpack-plans/types.ts:60]: This property is what triggers the activation of the `--full-width` style variant I mentioned earlier.

**Review by monsieur-z — CHANGES_REQUESTED** [review]
It looks good to me code wise, and the pricing page behaves as expected. One test is failing, though.

**Review by monsieur-z — APPROVED** [review]
Looking good now! Thanks for taking the time to do this.

**Pull request merged**